python/ray/tests/test_top_level_api.py

```diff
@@ -56,45 +66,68 @@ def test_non_ray_modules():
 
 def test_dynamic_subpackage_import():
     # Test that subpackages are dynamically imported and properly cached.
+    script = """
+import sys
+import ray
 
-    # ray.data
-    assert "ray.data" not in sys.modules
-    ray.data
-    # Check that the package is cached.
-    assert "ray.data" in sys.modules
+# ray.data
+assert "ray.data" not in sys.modules
+ray.data
+# Check that the package is cached.
+assert "ray.data" in sys.modules
 
-    # ray.workflow
-    assert "ray.workflow" not in sys.modules
-    ray.workflow
-    # Check that the package is cached.
-    assert "ray.workflow" in sys.modules
+# ray.workflow
+assert "ray.workflow" not in sys.modules
+ray.workflow
+# Check that the package is cached.
+assert "ray.workflow" in sys.modules
 
-    # ray.data
-    assert "ray.autoscaler" not in sys.modules
-    ray.autoscaler
-    # Check that the package is cached.
-    assert "ray.autoscaler" in sys.modules
+# ray.data
+assert "ray.autoscaler" not in sys.modules
+ray.autoscaler
+# Check that the package is cached.
+assert "ray.autoscaler" in sys.modules
+"""
+    _run_script_in_subprocess(script)
```

Comment:
You can call `run_string_as_driver`?

Reply:
oh i somehow thought it created a ray session - but yes, u r right.